test/commands/config/set.test.js

```diff
@@ -0,0 +1,273 @@
+				.it('should set api.nodes to empty array', () => {
+					const newConfig = {
+						...defaultConfig,
+						api: {
+							network: defaultConfig.api.network,
+							nodes: [],
+						},
+					};
+					return expect(config.setConfig).to.be.calledWith(
+						defaultDir,
+						newConfig,
+					);
+				});
+
+			setupStub()
+				.stdout()
+				.command(['config:set', 'api.nodes', 'ws://hostname'])
+				.catch(error =>
+					expect(error.message).to.contain(
+						'Node URLs must include a supported protocol',
+					),
+				)
+				.it(
+					'should throw error when api.nodes value is not supported protocol',
+				);
+
+			setupStub()
+				.stdout()
+				.command(['config:set', 'api.nodes', 'http://'])
+				.catch(error =>
+					expect(error.message).to.contain(
+						'Node URLs must include a supported protocol',
+					),
+				)
+				.it('should throw error when api.nodes value does not have host name');
+
+			setupStub()
+				.stdout()
+				.stub(config, 'getConfig', sandbox.stub().returns({}))
+				.command(['config:set', 'api.nodes', 'http://hostname'])
+				.catch(error =>
+					expect(error.message).to.contain(
+						'It looks like your configuration file is corrupted. Please check the file at',
+					),
+				)
+				.it('should throw error when config file is currupted');
+		});
+
+		describe('config:set json variable', () => {
+			setupStub()
+				.stdout()
+				.command(['config:set', 'json', 'true'])
+				.it('should json to provided value', () => {
+					const newConfig = {
+						...defaultConfig,
+						json: true,
+					};
+					return expect(config.setConfig).to.be.calledWith(
+						defaultDir,
+						newConfig,
+					);
+				});
+
+			setupStub()
+				.stub(config, 'setConfig', sandbox.stub().returns(false))
+				.stdout()
+				.command(['config:set', 'json', 'true'])
+				.catch(error =>
+					expect(error.message).to.contain(
+						'Config file could not be written: your changes will not be persisted',
+					),
+				)
+				.it('should throw an error when setConfig fails');
+
+			setupStub()
+				.stdout()
+				.command(['config:set', 'json', 'truely'])
+				.catch(error =>
+					expect(error.message).to.contain('Value must be a boolean.'),
+				)
+				.it('should throw error when json value is not boolean');
+		});
+
+		describe('config:set name variable', () => {
+			setupStub()
+				.stdout()
+				.command(['config:set', 'name', 'new name'])
+				.it('should set name to provided value', () => {
+					const newConfig = {
+						...defaultConfig,
+						name: 'new name',
+					};
+					return expect(config.setConfig).to.be.calledWith(
+						defaultDir,
+						newConfig,
+					);
+				});
+		});
+		describe('config:set api.network variable', () => {
+			const validNethash =
+				'198f2b61a8eb95fbeed58b8216780b68f697f26b849acf00c8c93bb9b24f783d';
+			setupStub()
+				.stdout()
+				.command(['config:set', 'api.network', validNethash])
+				.it(
+					'should throw error when api.network value is not valid hex string',
```

Comment:
`is unknown and not a valid hex string` but this description does not match the test.